akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala

```diff
@@ -14,30 +17,35 @@ import akka.annotation.{ DoNotInherit, InternalApi }
  * If the interceptor does keep mutable state care must be taken to create the instance in a `setup` block
  * so that a new instance is created per spawned actor rather than shared among actor instance.
  *
+ * @param interceptMessageClass Allows for applying the interceptor only to certain message types.
+ *                              If the message is not of this class or a subclass thereof it will
+ *                              bypass the interceptor and be continue to the inner behavior untouched.
  * @tparam O The outer message type – the type of messages the intercepting behavior will accept
+ * @tparam M The middle message type – the type of messages the interceptor cares about. Allows for
+ *           applying the interceptor only to certain message types.
+ *           If the message is not of this class or a subclass thereof it will
+ *           bypass the interceptor and be continue to the inner behavior untouched.
  * @tparam I The inner message type - the type of message the wrapped behavior accepts
```

Comment:
`Outer` and `Inner` are better than `O` and `I`.

Reply:
ok, I'll change that

Reply:
fixed